runtime/wasm/src/module/mod.rs

```diff
@@ -603,196 +610,62 @@ impl ModuleImportResolver for EnvModuleResolver {
     }
 }
 
-/// Store module resolver
-pub struct StoreModuleResolver;
+pub struct ModuleResolver;
 
-impl ModuleImportResolver for StoreModuleResolver {
-    fn resolve_func(&self, field_name: &str, _signature: &Signature) -> Result<FuncRef, Error> {
+impl ModuleImportResolver for ModuleResolver {
+    fn resolve_func(&self, field_name: &str, signature: &Signature) -> Result<FuncRef, Error> {
+        let signature = signature.clone();
         Ok(match field_name {
-            "set" => FuncInstance::alloc_host(
-                Signature::new(&[ValueType::I32, ValueType::I32, ValueType::I32][..], None),
```

Comment:
With these signatures gone (which saves a ton of boilerplate, so that's nice), is there still something that asserts that the call signature matches the function implementation signature?

Reply:
Yes, wasmi itself will error if you return the wrong signature (which makes me wonder what's the point if it knows the signature already...)